mm/2s2h/Enhancements/Trackers/ItemTracker/ItemTrackerSettings.cpp

```diff
@@ -0,0 +1,695 @@
+        windowObject.windowOpacity = opacity;
+    }
+    ImGui::PopID();
+    UIWidgets::Separator();
+}
+
+void ApplyRandoItemGroup(std::string listName) {
+    std::tuple<int16_t, int16_t, int16_t> list = randoItemLists.at(listName);
+    TrackerItemListObject itemObject = {
+        .windowName = listName,
+        .columnLength = std::get<2>(list),
+        .windowScale = 1.0f,
+        .windowOpacity = 0.5f,
+    };
+    for (int i = std::get<0>(list); i <= std::get<1>(list); i++) {
+        itemObject.itemList.push_back(i);
+    }
+
+    BenGui::mItemTrackerWindow->randoItemWindows.push_back(itemObject);
+}
+
+void ApplyDefaultItemGroup(std::string listName) {
+    std::tuple<int16_t, int16_t, int16_t> list = defaultItemLists.at(listName);
+    TrackerItemListObject itemObject = {
+        .windowName = listName,
+        .columnLength = std::get<2>(list),
+        .windowScale = 1.0f,
+        .windowOpacity = 0.5f,
+    };
+
+    if (listName == "Bottles" || listName == "Tokens" || listName == "Stray Fairies" || listName == "Dungeon") {
+        int16_t baseValue = std::get<0>(list);
+        int16_t maxValue = std::get<1>(list);
+        for (int j = baseValue; j <= maxValue; j++) {
+            itemObject.itemList.push_back(j);
+        }
+    } else {
+        std::pair<uint32_t, uint32_t> range = GetItemMapRange(std::get<0>(list), std::get<1>(list));
+        for (int i = range.first; i <= range.second; i++) {
+            if (itemIdToItemNameMap[i].first == ITEM_WALLET_ADULT) {
+                itemObject.itemList.push_back(ITEM_MAGIC_JAR_SMALL);
+                itemObject.itemList.push_back(ITEM_HEART_CONTAINER);
+            }
+            itemObject.itemList.push_back(itemIdToItemNameMap[i].first);
+        }
+    }
+    BenGui::mItemTrackerWindow->namedItemWindows.push_back(itemObject);
+}
+
+void ApplyDefaultItemPreset() {
+    BenGui::mItemTrackerWindow->namedItemWindows.clear();
+
+    for (int key = 0; key < defaultItemLists.size(); key++) {
+        if (listOrder[key] == "Rando") {
+            continue;
+        }
+        ApplyDefaultItemGroup(listOrder[key]);
+    }
+}
+
+void SaveItemTrackerLayout() {
+    std::vector<std::vector<TrackerItemListObject>*> trackerWindows = {
+        &BenGui::mItemTrackerWindow->namedItemWindows,
+        &BenGui::mItemTrackerWindow->randoItemWindows,
+    };
+
+    auto allConfig = Ship::Context::GetInstance()->GetConfig()->GetNestedJson();
+    if (allConfig.find("UConfs") == allConfig.end()) {
+        allConfig["UConfs"] = nlohmann::json::object();
+        Ship::Context::GetInstance()->GetConfig()->SetBlock("UConfs", allConfig["UConfs"]);
+        Ship::Context::GetInstance()->GetConfig()->Save();
+    }
+    allConfig["UConfs.ItemTrackerLayout"] = nlohmann::json::object();
+
+    auto& itemTrackerConfig = allConfig["UConfs.ItemTrackerLayout"];
+    itemTrackerConfig = nlohmann::json::object();
```

Comment:
Per the most recent discussion, I think we should just have this be its own top-level item.
```suggestion
    auto& itemTrackerConfig = allConfig["ItemTrackerLayout"] = nlohmann::json::object();
```